### PR TITLE
[DOCS-38] Work with streams via SDK

### PR DESCRIPTION
**Pull request by juhah**
Added "Work with streams via SDKs" section.

### CODE REVIEW

File: app/src/newdocs/content/code/streams.js [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+
+export const AuthJavaClient = 'StreamrClient client = new StreamrClient(new EthereumAuthenticationMethod(myEthereumPrivateKey));'
+
+export const SubscribeJavascriptClient =
+`// Subscribe to a stream
+const subscription = client.subscribe(
+    {
+        stream: 'MY-STREAM-ID',
+    },
+    function(message) {
```
Comment (timoxley) [inline app/src/newdocs/content/code/streams.js]: weird formatting + use arrows?

File: app/src/newdocs/content/code/streams.js [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+        // This function will be called when new messages occur
+        console.log(JSON.stringify(message))
```
Comment (timoxley) [inline app/src/newdocs/content/code/streams.js]: why JSON.stringify before logging?

File: app/src/newdocs/content/code/streams.js [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+`// Here is the event we'll be sending
+const msg = {
+    hello: 'world',
+    random: Math.random()
+}
+
+// Publish the event to the Stream
+client.publish('MY-STREAM-ID', msg)
+    .then(() => console.log('Sent successfully: ', msg))
+    .catch((err) => console.error(err))`
```
Comment (timoxley) [inline app/src/newdocs/content/code/streams.js]: IMO ignore error handling

File: app/src/newdocs/content/streams.mdx [hunk 4]
```diff
@@ -19,7 +33,61 @@ import docsStyles from '$newdocs/components/DocsLayout/docsLayout.pcss'
...
+<CodeSnippet language='javascript'>{AuthJavascriptClient}</CodeSnippet>
+
+**Java**
+
+<CodeSnippet language='java'>{CreateJavaClient}</CodeSnippet>
+
+To authenticate with an Ethereum account, a private key must be provided. The library will automatically initiate a challenge-response protocol to allow you to prove that you own the Ethereum private key without revealing it. You will be identified with the associated Ethereum public address.
+
+<CodeSnippet language='java'>{AuthJavaClient}</CodeSnippet>
+
```
Comment (timoxley) [inline app/src/newdocs/content/streams.mdx]: Perhaps should drop all of this auth content and put it in the auth section i.e. there's more about auth here than actually reading/writing to streams.

Comment (timoxley) [reply app/src/newdocs/content/streams.mdx]: Also note this content is near identical to what's added in https://github.com/streamr-dev/streamr-platform/pull/557/files#diff-923b7bd635c3f3e482653871e4d467f4 so probably should think about what belongs where.

IMO this section should maybe be a "deeper" dive into the intro content linked above, so perhaps it covers things like resends & realtime vs historical data? Error handling? rate limiting? etc

File: app/src/newdocs/content/streams.mdx [hunk 5]
```diff
@@ -19,7 +33,61 @@ import docsStyles from '$newdocs/components/DocsLayout/docsLayout.pcss'
 <ScrollableAnchor id="work-with-streams-via-sdks"><div>
 
 ## Work with streams via SDKs
---content--
+
+The easiest way to work with streams is to use the JavaScript client, which works in `node.js` as well as the browser. See the <Link to={links.newdocs.SDKs}>SDKs section</Link> for the official and community-maintained client libraries available for various languages:
+
+- **JavaScript** client works in the browser as well as node.js.
+- **Java** client is work-in-progress but already implements the most important set of functionality.
```
Comment (timoxley) [inline app/src/newdocs/content/streams.mdx]: It should be obvious but might be worth mentioning the java client is server-side only?

File: app/src/newdocs/content/streams.mdx [hunk 6]
```diff
@@ -19,7 +33,61 @@ import docsStyles from '$newdocs/components/DocsLayout/docsLayout.pcss'
...
+
+Client libraries for other languages are on the <a href={links.community.trello} target="_blank" rel="nofollow noopener noreferrer">roadmap</a>. If there isn't a client library available for your language, you can dive into the details of the [websocket protocol](https://github.com/streamr-dev/streamr-client-protocol-js/blob/master/PROTOCOL.md)
```
Comment (timoxley) [inline app/src/newdocs/content/streams.mdx]: > If there isn't a client library available for your language

maybe:

> If a client library isn't available for your language

File: app/src/newdocs/content/streams.mdx [hunk 7]
```diff
@@ -19,7 +33,61 @@ import docsStyles from '$newdocs/components/DocsLayout/docsLayout.pcss'
...
---content--
+
+The easiest way to work with streams is to use the JavaScript client, which works in `node.js` as well as the browser. See the <Link to={links.newdocs.SDKs}>SDKs section</Link> for the official and community-maintained client libraries available for various languages:
+
+- **JavaScript** client works in the browser as well as node.js.
+- **Java** client is work-in-progress but already implements the most important set of functionality.
+
+Client libraries for other languages are on the <a href={links.community.trello} target="_blank" rel="nofollow noopener noreferrer">roadmap</a>. If there isn't a client library available for your language, you can dive into the details of the [websocket protocol](https://github.com/streamr-dev/streamr-client-protocol-js/blob/master/PROTOCOL.md) 
+
+If you'd like to contribute a client library and get it listed here, please get in touch on [Telegram](https://t.me/streamrdata), [Reddit](https://www.reddit.com/r/streamr/) or [email](mailto:contact@streamr.com)!
```
Comment (timoxley) [inline app/src/newdocs/content/streams.mdx]: minor but reddit, telegram & general contact links do exist in `links.js`

File: app/src/newdocs/content/streams.mdx (hunk 6 above)
Comment (timoxley) [reply app/src/newdocs/content/streams.mdx]: also I have concerns about linking to `master` vs linking to a commit, if `PROTOCOL.md` ever moves then this link breaks, while a link like: https://github.com/streamr-dev/streamr-client-protocol-js/blob/bb2f0168bcff454b5c458e275e9c2de0dee342f9/PROTOCOL.md will always point at the same place, though downside there is that it won't automatically update as `PROTOCOL.md` is updated. It's a tradeoff but I  hit broken links to things in blob/master all the time and it's very annoying.

File: app/src/newdocs/content/code/streams.js [hunk 8]
```diff
@@ -0,0 +1,62 @@
...
+
+export const PublishJavaClient =
+`// Create the message payload, which is represented as a Map
+Map<String, Object> msg = new LinkedHashMap<>();
+msg.put("foo", "bar");
+msg.put("random", Math.random());
+
+// Then publish it!
+client.publish(stream, msg);
+All events are timestamped. The above example assigns the current timestamp to the new event, but you can also provide a timestamp explicitly:
```
[the remark this answers is not in the document]
Comment (juhah) [reply app/src/newdocs/content/code/streams.js]: Removed & kept simple.

File: app/src/newdocs/content/code/streams.js (hunk 1 above)
Comment (juhah) [reply app/src/newdocs/content/code/streams.js]: This is copy-paste from the repo page but changed it now

File: app/src/newdocs/content/code/streams.js (hunk 2 above)
Comment (juhah) [reply app/src/newdocs/content/code/streams.js]: Copy-paste code from the repo, I imagine to signify that the response is an object. Left it in.

File: app/src/newdocs/content/streams.mdx (hunk 4 above)
Comment (juhah) [reply app/src/newdocs/content/streams.mdx]: Ok, I left out the authentication and kept only the simple examples here. I feel the deep dive should happen in the SDK section (or leave it for the Github repos). But maybe those examples could be added here as well.

File: app/src/newdocs/content/streams.mdx (hunk 5 above)
Comment (juhah) [reply app/src/newdocs/content/streams.mdx]: There is no mention of it in the lib repo so I left it out from here as well.

File: app/src/newdocs/content/streams.mdx (hunk 6 above)
Comment (juhah) [reply app/src/newdocs/content/streams.mdx]: Yep, permanent link is perhaps better.

File: app/src/newdocs/content/streams.mdx (hunk 5 above)
Comment (timoxley) [reply app/src/newdocs/content/streams.mdx]: :+1:

**Pull request merged**